app/scripts/controllers/rif/index.js

```diff
@@ -16,38 +18,121 @@ import ComposableObservableStore from './../../lib/ComposableObservableStore'
  */
 export default class RifController {
   constructor (props) {
-    if (!props.preferencesController) {
-      throw new Error('PreferencesController has to be present');
+    if (!props.metamaskController) {
+      throw new Error('MetamaskController has to be present');
     }
-
-    if (!props.networkController) {
-      throw new Error('NetworkController has to be present');
-    }
-
-    if (!props.transactionController) {
-      throw new Error('TransactionController has to be present');
-    }
-
-    this.preferencesController = props.preferencesController;
-    this.networkController = props.networkController;
-    this.transactionController = props.transactionController;
-    this.web3 = new Web3(this.networkController._provider);
+    this.metamaskController = props.metamaskController;
+    this.web3 = new Web3(this.metamaskController.networkController._provider);
 
     const initState = props.initState || {};
 
     this.rnsManager = new RnsManager({
       initState: initState.RnsManager,
-      preferencesController: this.preferencesController,
-      networkController: this.networkController,
-      transactionController: this.transactionController,
+      preferencesController: this.metamaskController.preferencesController,
+      networkController: this.metamaskController.networkController,
+      transactionController: this.metamaskController.txController,
+      web3: this.web3,
+    });
+
+    this.luminoManager = new LuminoManager({
+      initState: initState.LuminoManager,
+      web3: this.web3,
+      keyringController: this.metamaskController.keyringController,
+      preferencesController: this.metamaskController.preferencesController,
+      networkController: this.metamaskController.networkController,
+      transactionController: this.metamaskController.txController,
+    });
+
+    this.luminoExplorer = new LuminoExplorer({
```

Comment:
I think we could move this inside the lumino manager, since the lumino manager already have all the checks when lumino is available to use and also LuminoExplorer should be part of lumino so it should be inside the manager or another way is to change the name to be like LuminoExplorerManager or something like that.

Reply:
Yes, y moved this in last commit to the luminomanager! 
Check it out now!